cmd/livegrep-fetch-reindex/main.go

```diff
@@ -166,8 +177,12 @@ if test "$1" = "get"; then
 fi
 `)
 
-func callGit(program string, args []string, username string, password string) error {
+// calls cmd.Run() if returnOutput is false
+// and cmd.Output() otherwise
+// always returns an out []byte, but it will always be nill if returnOutput is false
+func callGit(program string, args []string, username string, password string, returnOutput bool) ([]byte, error) {
```

Comment:
Let's provide two wrappers around this that give a nicer interface than the bool and maybe-`nil` output. I'm thinking about Python's `check_call`/`check_output` distinction here for some inspiration.

We might call them

```
func callGit(program string, args []string, username, password string) error
func callGitGetOutput(program string, args []string, username, password string) ([]byte, error)
```

This will make call sites much clearer.

We can keep one implementation -- rename the existing function to `callGitInternal` or something, and then pass appropriate values for `output` from the wrappers.

Reply:
Nice, I like that. I've gone ahead and implemented this pretty much exactly as you described, works great.